### ISSUE TITLE
`<chrono>`: `time_zone::get_info` fails when the current code page cannot represent the localized time zone short ID.

### ISSUE THREAD

**Issue by strega-nil-ms** [microsoft/STL#3158]
See:

- Issue #3097
- PR #3102
- PR #3122 

It seems likely that the result we want is for, if the abbreviation fails to convert, to instead attempt to format `UTC+{}` or `UTC-{}`.

**Comment by OptimumCpp** [comment]
If you are still there, the problem is still present. Has any thing changed during recent builds?

**Comment by strega-nil-ms** [comment]
I don't believe so; we need someone who has time to do this, to actually do the work involved; unfortunately, none of the members of the team have had the time.

**Comment by jbrezina** [comment]
I have this piece of code:
```csharp
const std::chrono::seconds::rep utcSeconds{ std::chrono::duration_cast<std::chrono::seconds>(std::chrono::utc_clock::now().time_since_epoch()).count() };
```
and it throws the exceptions saying _The operation completed successfully_. The exception is thrown on Windows Server 2012 R2 Datacenter. It works on Windows 11 Pro 22H2. When I use _system_clock_ instead of _utc_clock_ then the code works on the server edition, but when I compare the _utc_clock_ and _system_clock_ result on Windows 11 there is a 26s difference.
Is this somehow relevant or should I open new issue?

**Comment by OptimumCpp** [comment]
There's a visual studio issue, which is partially solved(not actually). It
is a priblem about locale and mostly affects middle-eastern platforms. You
need to chage locale to english before calling timezone functions. Link to
issue:
https://github.com/microsoft/STL/issues/3097

در تاریخ جمعه ۲۹ سپتامبر ۲۰۲۳،‏ ۱۰:۵۸ jbrezina ***@***.***>
نوشت:
